File: 2021/10/20/MinDeletionSize.java
/**
 * https://leetcode.com/problems/delete-columns-to-make-sorted/
 * @author Andrew Jarombek
 * @since 10/20/2021
 */

public class MinDeletionSize {
    public int minDeletionSize(String[] strs) {
        int count = 0;

        for (int column = 0; column < strs[0].length(); column++) {
            int row = 1;

            while (row < strs.length && strs[row].charAt(column) >= strs[row - 1].charAt(column)) {
                row++;
            }

            if (row != strs.length) {
                count++;
            }
        }

        return count;
    }
}

File: 2022/05/27/minDeletionSize.ts
/**
 * https://leetcode.com/problems/delete-columns-to-make-sorted/
 * @author Andrew Jarombek
 * @since 5/27/2022
 */

function minDeletionSize(strs: string[]): number {
    let count = 0;

    for (let i = 0; i < strs[0].length; i++) {
        let prev = '@';

        for (let str of strs) {
            const c = str.charAt(i);

            if (c < prev) {
                count++;
                break;
            } else {
                prev = c;
            }
        }
    }

    return count;
}